roomserver/internal/input/input_events.go

```diff
@@ -228,10 +302,115 @@ func (r *Inputer) processRoomEvent(
 	return nil
 }
 
+func (r *Inputer) checkForMissingAuthEvents(
```

Comment:
This needs documentation. It's not clear by the function name what this is trying to do. It doesn't return anything either, so I'm assuming it has some side-effects.